src/ai/backend/agent/server.py

```diff
@@ -935,9 +936,18 @@ async def _push_image(reporter: ProgressReporter) -> None:
 
     @rpc_function_v2
     @collect_error
-    async def purge_images(self, images: list[str]) -> PurgeImageResponses:
-        log.info("rpc::purge_images(images:{0})", images)
-        return await self.agent.purge_images(images)
+    async def purge_images(
+        self, image_canonicals: list[str], force: bool, noprune: bool
+    ) -> PurgeImagesResp:
+        log.info(
+            "rpc::purge_images(images:{0}, force:{1}, noprune:{2})",
+            image_canonicals,
+            force,
+            noprune,
+        )
+        return await self.agent.purge_images(
+            PurgeImagesReq(images=image_canonicals, force=force, noprune=noprune)
+        )
```

Comment:
this wasn't the intention, but it seems to be a minor detail that keeps delaying things, so I'll let it slide this time.